src/foam/nanos/controller/ApplicationController.js

```diff
@@ -319,9 +319,9 @@ foam.CLASS({
       this.clientPromise.then(async function(client) {
         self.setPrivate_('__subContext__', client.__subContext__);
 
-        await self.fetchSubject();
         await client.translationService.initLatch;
         self.installLanguage();
+        await self.fetchSubject();
```

Comment:
After we get the subject, if their locale is different than foam.locale, then we need to re-installLanguage() again.